src/amo/reducers/home.js

```diff
@@ -66,40 +130,60 @@ type ApiAddonsResponse = {|
   results: Array<ExternalAddonType>,
 |};
 
-type LoadHomeAddonsParams = {|
+type LoadHomeDataParams = {|
   collections: Array<Object | null>,
+  heroShelves: ExternalHeroShelvesType,
   shelves: { [shelfName: string]: ApiAddonsResponse },
 |};
 
-type LoadHomeAddonsAction = {|
-  type: typeof LOAD_HOME_ADDONS,
-  payload: LoadHomeAddonsParams,
+type LoadHomeDataAction = {|
+  type: typeof LOAD_HOME_DATA,
+  payload: LoadHomeDataParams,
 |};
 
-export const loadHomeAddons = ({
+export const loadHomeData = ({
   collections,
+  heroShelves,
   shelves,
-}: LoadHomeAddonsParams): LoadHomeAddonsAction => {
+}: LoadHomeDataParams): LoadHomeDataAction => {
   invariant(collections, 'collections is required');
   invariant(shelves, 'shelves is required');
 
   return {
-    type: LOAD_HOME_ADDONS,
+    type: LOAD_HOME_DATA,
     payload: {
       collections,
+      heroShelves,
       shelves,
     },
   };
 };
 
-type Action = FetchHomeAddonsAction | LoadHomeAddonsAction | SetClientAppAction;
+type Action = FetchHomeDataAction | LoadHomeDataAction | SetClientAppAction;
 
 const createInternalAddons = (
   response: ApiAddonsResponse,
 ): Array<AddonType> => {
   return response.results.map((addon) => createInternalAddon(addon));
 };
 
+export const createInternalHeroShelves = (
+  heroShelves: ExternalHeroShelvesType,
+): HeroShelvesType => {
+  const { primary, secondary } = heroShelves;
+
+  return {
+    primary: {
+      gradient: primary.gradient,
+      featuredImage: primary.featured_image,
+      description: primary.description,
+      addon: primary.addon ? createInternalAddon(primary.addon) : undefined,
+      external: primary.external,
```

Comment:
I'd use `|| undefined` to be extra cautious, but maybe it's not needed.

Reply:
Sure, why not? 😄